internal/cli/default_setter_opts.go

```diff
@@ -211,45 +202,82 @@ func (opts *DefaultSetterOpts) OrgExists(id string) bool {
 // If it fails or there are no organizations to show we fallback to ask for org by ID.
 // If only one organization, select it by default without prompting the user.
 func (opts *DefaultSetterOpts) AskOrg() error {
-	oMap, oSlice, err := opts.orgs()
+	return opts.askOrgWithFilter("")
+}
+
+func (opts *DefaultSetterOpts) askOrgWithFilter(filter string) error {
+	orgs, err := opts.orgs(filter)
 	if err != nil {
+		applyFilter := false
 		var target *atlas.ErrorResponse
 		switch {
 		case errors.Is(err, errNoResults):
-			_, _ = fmt.Fprintln(opts.OutWriter, "You don't seem to have access to any organization")
+			if filter == "" {
+				_, _ = fmt.Fprintln(opts.OutWriter, "You don't seem to have access to any organization")
+			} else {
+				_, _ = fmt.Fprintln(opts.OutWriter, "No organizations found matching the filter")
+				applyFilter = true
+			}
 		case errors.Is(err, errTooManyResults):
-			_, _ = fmt.Fprintf(opts.OutWriter, "You have access to more than %d organizations\n", resultsLimit)
+			_, _ = fmt.Fprintf(opts.OutWriter, "Looks like there are more than %d organizations\n", resultsLimit)
+			applyFilter = true
 		case errors.As(err, &target):
 			_, _ = fmt.Fprintf(opts.OutWriter, "There was an error fetching your organizations: %s\n", target.Detail)
 		default:
 			_, _ = fmt.Fprintf(opts.OutWriter, "There was an error fetching your organizations: %s\n", err)
 		}
-		p := &survey.Confirm{
-			Message: "Do you want to enter the Org ID manually?",
-		}
-		manually := true
-		if err2 := telemetry.TrackAskOne(p, &manually); err2 != nil {
-			return err2
-		}
-		opts.AskedOrgsOrProjects = true
-		if manually {
-			p := prompt.NewOrgIDInput()
-			return telemetry.TrackAskOne(p, &opts.OrgID, survey.WithValidator(validate.OptionalObjectID))
+
+		if applyFilter {
+			_, _ = fmt.Fprintln(opts.OutWriter, "Type to filter or leave empty to input ID manually")
+			filterPrompt := &survey.Input{
+				Message: "Org Name filter:",
+				Help:    "Beginning of an Organization name to filter results.",
+			}
+			filterErr := telemetry.TrackAskOne(filterPrompt, &filter)
+			if filterErr != nil {
+				return filterErr
+			}
+			if filter != "" {
+				return opts.askOrgWithFilter(filter)
+			}
+			_, _ = fmt.Fprintln(opts.OutWriter, "No filter provided")
 		}
-		_, _ = fmt.Fprint(opts.OutWriter, "Skipping default organization setting\n")
-		return nil
+
+		return opts.manualOrgID()
 	}
 
-	if len(oSlice) == 1 {
-		opts.OrgID = oMap[oSlice[0]]
+	return opts.selectOrg(orgs)
+}
+
+func (opts *DefaultSetterOpts) manualOrgID() error {
+	p := &survey.Confirm{
+		Message: "Do you want to enter the Org ID manually?",
+	}
+	manually := true
+	if err2 := telemetry.TrackAskOne(p, &manually); err2 != nil {
```

Comment:
[Q] Why is it `err2`? Why not `err`?

Reply:
Our linters for not allow variable hiding, `err` is already defined

Reply:
copy/paste